spec/datadog/tracing/configuration/dynamic_spec.rb

```diff
@@ -0,0 +1,81 @@
+# frozen_string_literal: true
+
+require 'rspec'
+
+RSpec.shared_examples 'tracing dynamic simple option' do |name:, env_var:, config_key:, value:, config_object: nil|
+  let(:option) { described_class.new }
+  let(:configuration_object) { config_object || Datadog.configuration.tracing }
+  let(:new_value) { value }
+  let(:old_value) { configuration_object.public_send(config_key) }
+
+  before do
+    configuration_object.reset!
+    old_value
+  end
+
+  describe '#call' do
+    subject(:call) { option.call(new_value) }
+
+    it "changes #{config_key} to #{value}" do
+      expect { call }.to change { configuration_object.public_send(config_key) }.from(old_value).to(value)
+    end
+
+    it "declares environment variable name as #{env_var}" do
+      expect(option.env_var).to eq(env_var)
+    end
+
+    it "declares option name as #{name}" do
+      expect(option.name).to eq(name)
+    end
+
+    context 'with nil value' do
+      before { call }
+
+      it "restores original value before dynamic configuration #{config_key}" do
+        expect { option.call(nil) }.to change { configuration_object.public_send(config_key) }.from(value).to(old_value)
+      end
+    end
+  end
+end
+
+RSpec.describe Datadog::Tracing::Configuration::Dynamic::LogInjectionEnabled do
+  include_examples 'tracing dynamic simple option',
+                   name: 'log_injection_enabled',
+                   env_var: 'DD_LOGS_INJECTION',
+                   config_key: :log_injection,
+                   value: false
+end
+
+RSpec.describe Datadog::Tracing::Configuration::Dynamic::TracingHeaderTags do
+  include_examples 'tracing dynamic simple option',
+                   name: 'tracing_header_tags',
+                   env_var: 'DD_TRACE_HEADER_TAGS',
+                   config_key: :header_tags,
+                   value: RSpec::Matchers::BuiltIn::Match.new(->(header_tags) { header_tags.to_s == 'my-header:my-tag' })
```

Comment:
I want to refactor `option :header_tags` from https://github.com/DataDog/dd-trace-rb/pull/2935/files#diff-e39a5f803ff255c0c4f1872deaae5df5cd8fb110947b92ce918a6f03b8e96e35 back into a simple value.
The idea of having a never-nil object to always perform HTTP header tag was attractive for the Rack and Sinatra integrations, but it made it super hard to handle it in configuration testing.

I don't want to do it immediately because it will take a bit of time, and it is an internal-only refactor, with no visible side-effects.